frontend/pages/hosts/details/HostDetailsPage/HostActionsDropdown/HostActionsDropdown.tests.tsx

```diff
@@ -381,4 +395,182 @@ describe("Host Actions Dropdown", () => {
       expect(screen.getByText("Delete")).toBeInTheDocument();
     });
   });
+
+  describe("Lock action", () => {
+    it("renders when the host is enrolled in mdm and the mdm is enabled", async () => {
+      const render = createCustomRenderer({
+        context: {
+          app: {
+            isPremiumTier: true,
+            isMdmEnabledAndConfigured: true,
+            isGlobalAdmin: true,
+            currentUser: createMockUser(),
+          },
+        },
+      });
+
+      const { user } = render(
+        <HostActionsDropdown
+          hostTeamId={null}
+          onSelect={noop}
+          hostStatus="online"
+          hostMdmEnrollmentStatus="On (automatic)"
+          mdmName="Fleet"
+          hostPlatform="darwin"
+          hostMdmDeviceStatus="unlocked"
+        />
+      );
+
+      await user.click(screen.getByText("Actions"));
+
+      expect(screen.getByText("Lock")).toBeInTheDocument();
+    });
+
+    it("does not render when the host is not enrolled in mdm", async () => {
+      const render = createCustomRenderer({
+        context: {
+          app: {
+            isPremiumTier: true,
+            isMdmEnabledAndConfigured: true,
+            isGlobalAdmin: true,
+            currentUser: createMockUser(),
+          },
+        },
+      });
+
+      const { user } = render(
+        <HostActionsDropdown
+          hostTeamId={null}
+          onSelect={noop}
+          hostStatus="online"
+          hostMdmEnrollmentStatus="Off"
+          mdmName="Fleet"
+          hostPlatform="darwin"
+          hostMdmDeviceStatus="unlocked"
+        />
+      );
+
+      await user.click(screen.getByText("Actions"));
+
+      expect(screen.queryByText("Lock")).not.toBeInTheDocument();
+    });
+
+    it("does not render when the host is not enrolled in a Fleet MDM solution", async () => {
+      const render = createCustomRenderer({
+        context: {
+          app: {
+            isPremiumTier: true,
+            isMdmEnabledAndConfigured: true,
+            isGlobalAdmin: true,
+            currentUser: createMockUser(),
+          },
+        },
+      });
+
+      const { user } = render(
+        <HostActionsDropdown
+          hostTeamId={null}
+          onSelect={noop}
+          hostStatus="online"
+          hostMdmEnrollmentStatus="On (automatic)"
+          mdmName="Non Fleet MDM"
+          hostPlatform="darwin"
+          hostMdmDeviceStatus="unlocked"
+        />
+      );
+
+      await user.click(screen.getByText("Actions"));
+
+      expect(screen.queryByText("Lock")).not.toBeInTheDocument();
+    });
+  });
+
+  describe("Unlock action", () => {
```

Comment:
do you mean the unlocked tests? it is a different test setup so that is why it is under this describe. I've gonna update the test messages a bit more to hopefully make it make more sense. after rereading it, its a bit unclear

Reply:
Oh I meant there's like 4-5 describe tests, I figured we can put the `describe("Unlock action",` test directly under the `describe("Lock action",` test

Reply:
NM! I see it, it felt weird reading but I now realized I misread the organization of these tests